### ISSUE TITLE
[/scan_raw] does not appear to be published yet

### ISSUE THREAD

**Issue by Blueicaro** [fmrico/book_ros2#29]
Hello, when I execute br2_tiago (ros2 launch br2_tiago sim.launch.py) and then I try ros2 topic echo /scan_raw I got the error thar is not published.
I checked the list of topic (ros2 topic list) and the scan_raw is not present. Here is the log 

[launch.log](https://github.com/fmrico/book_ros2/files/13583317/launch.log)

Any idea, thanks

/Jorge

**Comment by fmrico** [comment]
Hi @Blueicaro 

I see that your controllers are failing at startup. What distro are you using? What branch of this repo are you using?

Best

**Comment by Blueicaro** [comment]
Hello, I using Ubuntu 20.04.6 LTS

I using foxy-devel branch

B.R.

**Comment by Blueicaro** [comment]
Hello, I found the problem and I fixed it. The problem was with the dependence. I ran rosdep update as you said in the book. But Foxy is EOL (end of life), so rosdep skipped it, as you see:
Skip end-of-life distro "ardent"
Skip end-of-life distro "bouncy"
Skip end-of-life distro "crystal"
Skip end-of-life distro "dashing"
Skip end-of-life distro "eloquent"
Skip end-of-life distro "foxy"
Skip end-of-life distro "galactic"
Skip end-of-life distro "groovy"
Add distro "humble"
Skip end-of-life distro "hydro"
Skip end-of-life distro "indigo"
Add distro "iron"
Skip end-of-life distro "jade"
Skip end-of-life distro "kinetic"
Skip end-of-life distro "lunar"
Skip end-of-life distro "melodic"
Add distro "noetic"
Add distro "rolling"
updated cache in /home/jetson/.ros/rosdep/sources.cache
So I ran again with option --rosdistro=foxy which set ros distro to use.
After that, I ran rosdep install --from-paths src --ignore-src -r -y and I compiled every thing again with colcon build --symlink-path as you explains in the book
Now every is running nice

/Jorge
